Skip running TheRock CI on docs-only changes

dwiddows opened the pull request:
## Motivation

Make documentation updates easier, by reducing CI test requirements for pull requests that change only doc files.

This is already done for TheRock and rocm-libraries, this PR follows the pattern of rocm-libraries in https://github.com/ROCm/rocm-libraries/pull/2679/

## Technical Details

Add filtering to skip CI builds when PRs only change documentation files.

## JIRA ID

<!-- If applicable, mention the JIRA ID resolved by this PR (Example: Resolves SWDEV-12345). -->
<!-- Do not post any JIRA links here. -->

## Test Plan

Unit tests for skippable patterns and behaviors added to .github/scripts/tests/therock_configure_ci_test.py
Added mock support as in https://github.com/ROCm/rocm-libraries/pull/2679/files

## Test Result

(Checked) python .\therock_configure_ci_test.py passes all 10 tests locally. 
(Expected) PRs which only change doc files should skip Window CI build and pass visibly faster.

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

dwiddows commented on .github/scripts/therock_configure_ci.py:
```diff
@@ -81,6 +81,27 @@ def check_for_workflow_file_related_to_ci(paths: Optional[Iterable[str]]) -> boo
 
 
+def is_doc_only_change(args) -> bool:
+    """Check if only .md, .rst, docs/*, or doc/* files are changed."""
```
Just seen that there are some tests in /.github/scripts/tests/therock_configure_ci_test.py so this might be worth adding to. What suggestion would you make?

ScottTodd replied:
Yes add unit tests.

See https://github.com/ROCm/TheRock/blob/0887429e82614c8c5286b1090061cfc48b116990/build_tools/github_actions/tests/configure_ci_test.py#L46-L64

dwiddows replied:
Added tests for skippable paths and used MagicMock pattern modeled on https://github.com/ROCm/rocm-libraries/pull/2679/files

dwiddows replied:
(resolving, tests are implemented and copilot comment is now confusing)

ScottTodd commented on .github/scripts/therock_configure_ci.py:
```diff
@@ -123,7 +141,12 @@ def retrieve_projects(args):
+    set_github_output({
+        "projects": json.dumps(project_to_run),
+        "is_doc_only": "true" if is_doc_only else "false"
+    })
```
Let's use the same setup as we have in TheRock: https://github.com/ROCm/TheRock/blob/0887429e82614c8c5286b1090061cfc48b116990/build_tools/github_actions/configure_ci.py#L120-L149

* No need to overspecialize on "is_doc_only". There are other files that can't affect CI and can skip CI.
* No need to set an output like "is_doc_only" - the list of projects to build/test should be empty if the changes in the PR are judged to not impact CI

dwiddows replied:
Updated and followed the pattern in rocm-libraries, as per https://github.com/ROCm/rocm-libraries/pull/2679/files and discussion in [Gardening channel teams thread](https://teams.microsoft.com/l/message/19:f50d2d50fd094a8e8065b11a1cfd3496@thread.tacv2/1765410797722?tenantId=3dd8961f-e488-4e60-8e11-a82d994e183d&groupId=69e7275a-4d49-495d-8bd5-04fb519c8e9c&parentMessageId=1765405670322&teamName=AIG%20ROCm&channelName=Gardening%20-%20rocm-systems&createdTime=1765410797722)

Resolving this conversation, code has changed.

dwiddows committed:
Skip Windows CI for doc-only changes

Following the pattern from ROCm/rocm-libraries#2679, add logic to skip
CI builds when only documentation files are modified.

Changes:
- Add SKIPPABLE_PATH_PATTERNS for docs, markdown, and .gitignore files
- Return empty projects list when only skippable paths are modified
- No workflow changes needed - existing projects != '[]' check handles it
- Add unit tests for doc-filtering logic
- Fix existing tests with proper subprocess mocking

Reference: https://github.com/ROCm/rocm-libraries/pull/2679

dwiddows commented:
Thanks @ScottTodd and @geomin12 for comments.

Implementation is changed to follow the pattern of https://github.com/ROCm/rocm-libraries/pull/2679/files.

For full parity I could also include the `test_type` full or smoke tests switch used there - this would make this change slightly larger. Should I do that? Any other recommendations?

geomin12 commented:
> Thanks @ScottTodd and @geomin12 for comments.
> 
> Implementation is changed to follow the pattern of https://github.com/ROCm/rocm-libraries/pull/2679/files.
> 
> For full parity I could also include the `test_type` full or smoke tests switch used there - this would make this change slightly larger. Should I do that? Any other recommendations?

we can leave that out for now! When we get tests added for rocm-systems, we will want to run full tests regardless (as these are critical system items) to make sure nothing is broken

dwiddows committed:
Merge branch 'develop' into users/dwiddows/skip-windows-ci-for-doc-only-prs

geomin12 reviewed (approved):
fantastic, thanks for this update!
